pkg/applicationserver/applicationserver.go

```diff
@@ -104,7 +108,13 @@ var errListenFrontend = errors.DefineFailedPrecondition("listen_frontend", "fail
 
 // New returns new *ApplicationServer.
 func New(c *component.Component, conf *Config) (as *ApplicationServer, err error) {
-	ctx := log.NewContextWithField(c.Context(), "namespace", "applicationserver")
+	ctx := tracer.NewContext(c.Context(),
+		telemetry.FromContext(c.Context()).Tracer(packageName),
+	)
+	ctx, span := tracer.FromContext(ctx).Start(ctx, "New")
```

Comment:
I think that tracing `New` calls is a bit overkill - we should focus on RPCs instead for this iteration. Same in other spots.